src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,240 @@
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "ctp" + "<playerName>";
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+        if (!canCommandSenderUseCommand(COLONYTP))
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("Not happenin bro!!, ask an OP to TP you."));
+            return;
+        }
+
+        EntityPlayer playerToTeleport = null;
+
+        if(sender instanceof EntityPlayer)
+        {
+            playerToTeleport = (EntityPlayer) sender;
+        }
+
+        //If the arguments aren't empty, the sender probably wants to teleport another player.
+        if (args.length != 0 && isPlayerOpped(sender, "ctp"))
+        {
+            final World world = FMLCommonHandler.instance().getMinecraftServerInstance().getEntityWorld();
+            playerToTeleport =
+                    ServerUtils.getPlayerFromUUID(FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerProfileCache()
+                            .getGameProfileForUsername(args[0]).getId(), world);
+
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("TPin Player: "+ playerToTeleport.getName()));
+        }
+
+        if(playerToTeleport == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+        teleportPlayer(sender, playerToTeleport);
+    }
+
+    /**
+     * Method used to teleport the player.
+     * @param sender the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
+        {
+            attCounter++;
+            int x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+            int z = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+
+            //Search for a ground position
+            final BlockPos groundPosition = findLand(new BlockPos(x, STARTING_Y, z), sender.getEntityWorld());
+
+            //If no position found
+            if (groundPosition == null)
+            {
+                attCounter++;
+                continue;
+            }
+
+            boolean foundPosition = isPositionSafe(sender, groundPosition) && !findColony(groundPosition, sender.getEntityWorld());
```

Comment:
findColony should be called before that. Because colony is not dependant of y coordinate.

performance wise if you hit a colony you will iterate through many y this way